Source/UETorch/Private/UETorch.cpp

```diff
@@ -706,3 +706,43 @@ extern "C" bool CaptureDepthField(UObject* _this, const IntSize* size, void* dat
 	}
 	return true;
 }
+
+extern "C" bool SetActorPosition(AActor* object, double x, double y, double z) {
+	return object->SetActorLocation(FVector(x,y,z), false);
+}
+
+extern "C" bool SetActorRotation(AActor* object, double pitch, double roll, double yaw) {
+	return object->SetActorRotation(FRotator(pitch,roll,yaw));
+}
+
+extern "C" void SetActorVisible(AActor* object, bool visible) {
+	object->SetActorHiddenInGame(!visible);
+}
+
+extern "C" bool SetActorVelocity(AActor* object, double x, double y, double z) {
+	if(!object) {
+		printf("object doesn't exist\n");
+		return false;
+	}
+	UStaticMeshComponent *component = object->FindComponentByClass<UStaticMeshComponent>();
+	if(!component) {
+		printf("Object doesn't have StaticMeshComponent\n");
+		return false;
+	}
+	component->SetPhysicsLinearVelocity(FVector(x,y,z));
+	return true;
+}
+
```

Comment:
Just ignore that - the doc for SetPhysicsLinearVelocity is the same. If
you're going to add SetPhysicsLinearVelocity, then
SetPhysicsAngularVelocity is exactly the same.

You can add the force/impulse versions as well if you want. But not
necessary at the moment.

On Thu, May 26, 2016 at 3:19 PM, Mario Ynocente Castro <
notifications@github.com> wrote:

> In Source/UETorch/Private/UETorch.cpp
> <https://github.com/facebook/UETorch/pull/12#discussion_r64742957>:
>
> > +}
> > +
> > +extern "C" bool SetActorVelocity(AActor* object, double x, double y, double z) {
> > +	if(!object) {
> > +		printf("object doesn't exist\n");
> > +		return false;
> > +	}
> > +	UStaticMeshComponent *component = object->FindComponentByClass<UStaticMeshComponent>();
> > +	if(!component) {
> > +		printf("Object doesn't have StaticMeshComponent\n");
> > +		return false;
> > +	}
> > +	component->SetPhysicsLinearVelocity(FVector(x,y,z));
> > +	return true;
> > +}
> > +
>
> Do you have an idea about how this should be implemented the doc for
> SetPhysicsAngularVelocity
> <https://docs.unrealengine.com/latest/INT/API/Runtime/Engine/Components/UPrimitiveComponent/SetPhysicsAngularVelocity/index.html>
> suggests using AddTorque
> <https://docs.unrealengine.com/latest/INT/API/Runtime/Engine/Components/UPrimitiveComponent/AddTorque/index.html>
> or AddImpulse
> <https://docs.unrealengine.com/latest/INT/API/Runtime/Engine/Components/UPrimitiveComponent/AddImpulse/index.html>,
> also all the three are receiving a BoneName variable.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/facebook/UETorch/pull/12/files/1c6bb824ed712c410524f83e5c847a027aa9a354#r64742957>
>